Source/Core/DolphinQt/DiscordJoinRequestDialog.cpp

```diff
@@ -0,0 +1,90 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#ifdef USE_DISCORD_PRESENCE
+
+#include <QGridLayout>
+#include <QLabel>
+#include <QPixmap>
+#include <QPushButton>
+
+#include <discord-rpc/include/discord_rpc.h>
+
+#include "Common/HttpRequest.h"
+#include "Common/StringUtil.h"
+
+#include "DolphinQt/DiscordJoinRequestDialog.h"
+
+DiscordJoinRequestDialog::DiscordJoinRequestDialog(QWidget* parent, const char* id,
+                                                   const std::string& discord_tag,
+                                                   const char* avatar)
+    : QDialog(parent), m_user_id(id), m_close_timestamp(std::time(nullptr) + s_max_lifetime_seconds)
+{
+  setWindowTitle(tr("Request to Join Your Party"));
+  setWindowFlags(windowFlags() & ~Qt::WindowContextHelpButtonHint);
+
+  QPixmap avatar_pixmap;
+
+  if (avatar[0] != '\0')
+  {
+    const std::string avatar_endpoint =
+        StringFromFormat("https://cdn.discordapp.com/avatars/%s/%s.png", id, avatar);
+
+    Common::HttpRequest request;
+    Common::HttpRequest::Response response = request.Get(avatar_endpoint);
+
+    if (response.has_value())
+      avatar_pixmap.loadFromData(response->data(), static_cast<uint>(response->size()), "png");
+  }
+
+  CreateMainLayout(discord_tag, avatar_pixmap);
+  ConnectWidgets();
+}
+
+std::time_t DiscordJoinRequestDialog::GetCloseTimestamp() const
+{
+  return m_close_timestamp;
+}
+
+void DiscordJoinRequestDialog::CreateMainLayout(const std::string& discord_tag,
+                                                const QPixmap& avatar)
+{
+  m_main_layout = new QGridLayout;
+
+  m_invite_button = new QPushButton(QString::fromWCharArray(L"\u2714 Invite"));
+  m_decline_button = new QPushButton(QString::fromWCharArray(L"\u2716 Decline"));
```

Comment:
I don't think it matters what's more appropriate as they both a pain X. I'm not sure if Cancellation X is actually more appropriate, since it seems to be for closing a window instead of declining an invite but then again pressing it does close the window.